thunder/core/transforms.py

```diff
@@ -3189,10 +3196,11 @@ def recompute_saved_for_backward(fwd_trace: Trace, bwd_trace: Trace) -> tuple[Tr
             new_unpack = prims.unpack_sequence.bind(*unpack_args, output=new_saved_for_backward)
             new_bwd_trace.bound_symbols.append(new_unpack)
         elif idx == 6:
-            new_bwd_trace.bound_symbols.extend(producers)
-            new_bwd_trace.bound_symbols.append(bsym)
+            for p in producers:
+                new_bwd_trace.bound_symbols.append(p.from_bsym_swap_proxies(swap_map=swap_map))
+            new_bwd_trace.bound_symbols.append(bsym.from_bsym_swap_proxies(swap_map=swap_map))
         else:
-            new_bwd_trace.bound_symbols.append(bsym)
+            new_bwd_trace.bound_symbols.append(bsym.from_bsym_swap_proxies(swap_map=swap_map))
```

Comment:
Hmm apparently torch compile backend breaks with the rename:
```shell
 File "/opt/pytorch/lightning-thunder/thunder/core/langctxs.py", line 136, in _fn
    result = fn(*args, **kwargs)
  File "/opt/pytorch/lightning-thunder/thunder/__init__.py", line 233, in cache_info_wrapper
    res = fn(*args, **kwargs)
  File "/opt/pytorch/lightning-thunder/thunder/__init__.py", line 657, in get_computation_and_inputs
    computation_trc, backward_trc = split_forward_backward(computation_trc, cd, cs, *inps)
  File "/opt/pytorch/lightning-thunder/thunder/executors/torch_autograd.py", line 250, in split_forward_backward
    bw_extrace = transform_for_execution(
  File "/opt/pytorch/lightning-thunder/thunder/executors/passes.py", line 160, in transform_for_execution
    extrace = ex.fusion_pass(extrace)
  File "/opt/pytorch/lightning-thunder/thunder/executors/torch_compile.py", line 194, in fusion_pass
    fusedtrace = dce(fusedtrace)
  File "/opt/pytorch/lightning-thunder/thunder/core/transform_common.py", line 121, in dce
    flat_trace_outputs, _ = tree_flatten(trace.output)
  File "/opt/pytorch/lightning-thunder/thunder/core/trace.py", line 240, in output
    assert False, "Trace has no return, and so no output"
AssertionError: Trace has no return, and so no output
```

Even tho the trace does pass the asserts earlier and therefore it contains a return before going to the fusion pass. Let's fix that before merging this

Reply:
Could you please add a test first before attempting to fix the issue? CI hasn't detected anything.

Reply:
It doesn't seem to be a problem anymore for two main reasons:
- saved for backward remat is not in the code path anymore when using torch compile for the example that was showing up the issue.
- Can't reproduce the problem anymore even when using ThunderFX.

At this point  I'd rely on the test and mark this as solved for now.